### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by honza** [blevesearch/bleve#162]
I may get labelled a hater for this and I'm fine with that.

Bleve is an awesome project with a lot of potential which is only hindered by the lack of documentation. One is left to read the source code and outdated examples which frankly don't work as I'd expect.

It would be super awesome if there was a nice tutorial for bleve to do something non-trivial. The analysis tool is great for creating custom analyzers and filters. But how do I plug it all together?

Also, installation and building isn't exactly straight-forward and mostly undocumented. I'm sitting here, trying all kinds of combinations of build flags hoping to get it to work.

Yes, this is a ton of stuff in the wiki but it's a mess with few details about many important aspects of this great library.

I would love help with this stuff but as a beginner I'm not sure I can. Please take this as an encouragement and not an put-down. Thanks!

**Comment by mschoch** [comment]
> It would be super awesome if there was a nice tutorial for bleve to do something non-trivial. The analysis tool is great for creating custom analyzers and filters. But how do I plug it all together?

Yeah a tutorial would be nice to have.  We have examples that do non-trivial things, but nothing in a tutorial format walking a user through from point A to point B.

> Also, installation and building isn't exactly straight-forward and mostly undocumented. I'm sitting here, trying all kinds of combinations of build flags hoping to get it to work.

In general we've been evolving things to eliminate the need for build flags, at least for beginners.  The exceptions are:

- Using leveldb, not that complicated, need leveldb on your machine, and add build flag 'leveldb'
-  Using icu, not needed for most languages any more since we introduced our own text segmentation library.  Yes, advanced users may still need this.
-  Using cld2.  We should probably consider removing this.  At the moment its not good for anything other than adding a field with the best guess language.  We plan to maybe do more with this in the future, but I'd hate to thing people get hung up building this, when they likely don't need it.  Separately there is another effort to package up a gocld2 library, which extracts cld2 in place and allows for static linking directly through the go toolchain.  This would eliminate much of the pain here.

The best I can recommend is to continue to open issues for things that don't work as expected, and ask for help when you get stuck (google-group is best bet for this).  Better documentation is not likely to appear overnight without others helping to contribute.

**Comment by ktmn** [comment]
Couldn't agree more. Bleve is amazing but right now it's not friendly for beginners like me - I have no idea what I'm doing and the examples are pretty vague.

**Comment by lundin** [comment]
Yes i echo the tip to try the google-groups. very useful. I always trying to paste relevant code into the post or collect it into a gist.

Also the build option for ubuntu in the wiki is spot-on same for RHEL and yum manager as well. If you are on ubuntu, this is simply a: sudo apt-get install libstemmer-dev

The hardest part for me in the beginning was that i was on a mac and i didn't know how to get the libstemmer up and running. However follwing steps works (for me)

The library expects the C library to be installed system-wide and links against that.

clone from GIT https://github.com/zvelo/libstemmer
install cmake (brew install cmake)
unzip files and cd to folder
cmake .
make
sudo make install (installs .a .dylib shared lib and .h)

Then build with the libstemmer tag

Bleve is evolving and under development, we should help the developers with the wiki part the way i see it.
